pi5/board.go

```diff
@@ -113,19 +113,24 @@ func newBoard(
 		gpios:      map[string]*pinctrl.GPIOPin{},
 		interrupts: map[string]*pinctrl.DigitalInterrupt{},
 
-		chipSize: 0x30000,
-		pulls:    map[int]byte{},
+		pulls: map[int]byte{},
+	}
+
+	pinctrlCfg := pinctrl.Config{GPIOName: "gpio0", GPIOMemPath: "/dev/gpiomem0", DTBase: "/proc/device-tree", ChipSize: 0x30000}
+	if testingMode {
+		pinctrlCfg.TestPath = "./mock-device-tree"
 	}
 
 	// Note that this must be called before configuring the pull up/down configuration uses the
 	// memory mapped in this function.
-	if err := b.setupPinControl(testingMode); err != nil {
+	b.boardPinCtrl, err = pinctrl.SetupPinControl(pinctrlCfg, logger)
+	if err != nil {
 		return nil, err
 	}
 
 	// Initialize the GPIO pins
 	for newName, mapping := range gpioMappings {
-		b.gpios[newName] = pinctrl.CreateGpioPin(mapping, &b.vPage, logger)
+		b.gpios[newName] = b.boardPinCtrl.CreateGpioPin(mapping)
 	}
```

Comment:
thought about making this into a helper function on the boardPinCtrl struct, but I was unsure if this functionality would be generalized to other boards and how this functionality would interact with the raspberry-pi config/being about to name gpio pins